demos/demo15_test_LMRF.py

```diff
@@ -7,12 +7,12 @@
 import scipy
 import cuqi.operator
 #%%
+location = 0
 N = 2
-mean = np.zeros(N)
 prec = 1.0
 dom = 1
 BCs = 'neumann'
-x = cuqi.distribution.LMRF(mean, prec, BCs)
+x = cuqi.distribution.LMRF(location, prec, BCs, dim=N)
```

Comment:
Here a 1D version is obtained. I want to understand which combinations of location, dim, physical_dim and geometry are allowed, incl in which different ways one can set up a 2D XMRF. 

Can you do a scalar location/prec, a scalar dim and physical_dim = 1 or 2?
Can you do a scalar location/prec, no dim, and a physical_dim 1 or 2?
Can you do scalar location/prec, no dim, no physical_dim, and a 1D or 2D geometry?
Can you do vec location/prec, no dim, no physical_dim, and a 1D or 2D geometry?
etc

And if all conflicting sets of inputs are handled? 

Also what happens if geometry that doesn't make sense with XMRF is passed, like StepExpansion?